ast/functions.go

```diff
@@ -532,6 +538,39 @@ func (n *AggregateFuncExpr) updateSum() error {
 	return nil
 }
 
+func (n *AggregateFuncExpr) updateGroupConcat() error {
+	ctx := n.GetContext()
+	vals := make([]interface{}, 0, len(n.Args))
+	for _, a := range n.Args {
+		value := a.GetValue()
+		if value == nil {
+			return nil
+		}
+		vals = append(vals, value)
+	}
+	if n.Distinct {
+		d, err := ctx.distinctChecker.Check(vals)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if !d {
+			return nil
+		}
+	}
+	var buf bytes.Buffer
```

Comment:
We can put buffer in context, so we don't need to create a new buffer on every update.